src/psyclone/psyir/transformations/read_only_verify_trans.py

```diff
@@ -0,0 +1,138 @@
+   a region of code."
+'''
+
+from __future__ import absolute_import
+from psyclone.psyir.nodes import ReadOnlyVerifyNode, Schedule
+from psyclone.psyir.transformations.psy_data_trans import PSyDataTrans
+from psyclone.psyir.transformations.transformation_error \
+    import TransformationError
+
+
+class ReadOnlyVerifyTrans(PSyDataTrans):
+    '''This transformation inserts a ReadOnlyVerifyNode or a node derived
+    from ReadOnlyVerifyNode into the PSyIR of a schedule. At code creation
+    time this node will use the PSyData API to create code that will
+    verify if fields are modified.
+
+    After applying the transformation the Nodes marked for verification are
+    children of the ReadOnlyVerifyNode.
+    Nodes to verify can be individual constructs within an Invoke (e.g.
+    Loops containing a Kernel or BuiltIn call) or entire Invokes.
+
+    :param node_class: The Node class of which an instance will be inserted \
+        into the tree (defaults to ReadOnlyVerifyNode), but can be any \
+        derived class.
+    :type node_class: :py:class:`psyclone.psyir.nodes.ReadOnlyVerifyNode` or \
+        derived class
+
+    '''
+    from psyclone.psyir import nodes
+    from psyclone import psyGen
+    # The types of node that this transformation can enclose
+    valid_node_types = (nodes.Loop, psyGen.Kern, psyGen.BuiltIn,
+                        psyGen.Directive, nodes.Literal, nodes.Reference)
+
+    def __init__(self, node_class=ReadOnlyVerifyNode):
+        super(ReadOnlyVerifyTrans, self).__init__(node_class=node_class)
+
+    def __str__(self):
+        return ("Create a sub-tree of the PSyIR that has a "
+                "ReadOnlyVerifyNode at its root.")
+
+    @property
+    def name(self):
+        ''' Returns the name of this transformation as a string.'''
+        return "ReadOnlyVerifyTrans"
+
+    def validate(self, node_list, options=None):
+        '''Performs validation checks specific to read-only-based
+        transformations.
+
+        :param node_list: the list of Node(s) we are checking.
+        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if distributed memory is configured.
+        :raises TransformationError: if transformation is applied to a \
+                                     Kernel or a BuiltIn call without its \
+                                     parent Loop.
+        :raises TransformationError: if transformation is applied to a Loop \
+                                     without its parent Directive when \
+                                     optimisations are applied.
+        :raises TransformationError: if transformation is applied to an \
+                                     orphaned Directive without its parent \
+                                     Directive.
+        '''
+
```

Comment:
I don't think you need the blank lines before and after the comment.

Reply:
Done.